### PR TITLE
Adding Stacked Checkbox component

### PR DESCRIPTION
**Pull request by npfitz**
### Description
Adds a Stacked Checkbox component, which is just a specifically styled Mantine checkbox. Styled using static CSS selectors to hopefully make the migration to v7 easier in the future. This new checkbox is also set to be used in add / remove columns.

### How to verify
New Checkbox can be viewed in the Add / Remove columns panel (create a structured question and view viz settings) or found in storybook.

### Demo
image

### CODE REVIEW

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -14,5 +15,9 @@ export const parameters = {
 };
 
 export const decorators = [
-  renderStory => <ThemeProvider>{renderStory()}</ThemeProvider>,
+  renderStory => (
```
Comment (npfitz) [inline .storybook/preview.tsx:18]: The static class prefix by default is `mantine`. However, when using an emotion cache, it changes to the key of the emotion cache. By default our storybook previews didn't have a cache set, so the prefixes were different between storybook and the metabase app ('css' and 'emotion') respectively. This change brings those back in line.

**Review by ranquild — CHANGES_REQUESTED** [review]
We could introduce another Checkbox "variant" and put all styles properly in `Checkbox`. This would allow us to not override anything using static CSS selectors.

**Comment by npfitz** [comment]
Updated. Awesome Suggestion @ranquild

File: frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.styled.tsx [hunk 2]
```diff
@@ -76,5 +77,40 @@ export const getCheckboxOverrides = (): MantineThemeOverride["components"] => ({
...
+            left: "4px",
+            height: "100%",
+            width: "100%",
+            borderRadius: "4px",
+            zIndex: -1,
+            backgroundColor: "white",
+            boxSizing: "border-box",
+          },
+          "&:checked:after": {
+            border: `2px solid ${color("brand")}`,
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.styled.tsx:104]: `theme.colors.brand[1]`

File: frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.styled.tsx [hunk 3]
```diff
@@ -76,5 +77,40 @@ export const getCheckboxOverrides = (): MantineThemeOverride["components"] => ({
...
+        label: {
+          paddingLeft: labelPosition === "right" ? "0.75rem" : "0.5rem",
+        },
+        input: {
+          "&:after": {
+            content: "''",
+            border: `1px solid ${theme.colors.gray[4]}`,
+            position: "absolute",
+            top: "-4px",
+            left: "4px",
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.styled.tsx:95]: theme.spacing.xs

**Comment by mazameli** [comment]
Nice! This is looking good to me so far. 

Detail question: why in storybook is the `size` param a string input rather than a list of discrete options like `sm`, `md`, `lg`?

**Comment by npfitz** [comment]
> why in storybook is the size param a string input rather than a list of discrete options like sm, md, lg?
It should probably be a radio list, but that param will also accept an int as a number of pixels. Not sure how to make Storybook understand that, but I'll at least make the change you're suggesting here

**Review by ranquild — COMMENTED** [review]
Code LGTM, there is one lint error

**Review by ranquild — APPROVED** [review]

**Comment by ranquild** [comment]
@npfitz we use a radio for other components for `size`, we could do it here as well. Howerver I don't think that the component supports it right now

**Comment by npfitz** [comment]
Updated to use radio component for size 👍

**Comment by mazameli** [comment]
This may've been a preexisting issue with the `checkbox` component before this PR, and if so, I think we could solve it separately if we have to — but, when you add a description, the box is getting aligned incorrectly relative to the label. Here it is in Storybook, with the label on left and right:

image

image


And here it is in Mantine, with the label on both left and right. This is the desired alignment IMO:

image

image

**Comment by mazameli** [comment]
Beyond that, this PR looks good to me.

**Comment by npfitz** [comment]
Ohh, yeah. That was something I probably did in this PR. I'll see if I can get that fixed.

**Review by kdoh — APPROVED** [review]

**Pull request merged**